bin/improver-nbhood

```diff
@@ -189,7 +194,22 @@ def main():
         parser.wrong_args_error(
             'neighbourhood_shape=circular', 'input_mask_filepath')
 
+    if args.wind_direction:
+        if args.neighbourhood_output == "percentiles":
+            parser.error('Cannot generate percentiles from complex wind '
+                         'directions')
+        if args.neighbourhood_shape == "circular":
+            parser.error('Cannot process complex wind directions with '
+                         'circular neighbourhoods')
+        if args.apply_recursive_filter:
```

Comment:
As I understand it, you would want to run the recursive filter on complex numbers rather than degrees for the same reason you want to run neighbourhood processing and realization collapse on complex numbers.

I haven't tested whether recursive filtering will run using complex numbers - I'd not realised that was in scope until starting work on the CLI.  I could look into recursive filtering, and fix or create follow on work if necessary?

Reply:
Not sure it is in scope, I just hadn't appreciated the reasoning for excluding it.